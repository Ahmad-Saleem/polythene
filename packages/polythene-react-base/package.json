{
  "name": "polythene-react-base",
  "version": "1.0.0-rc.8",
  "description": "",
  "main": "dist/polythene-react-base.js",
  "module": "dist/polythene-react-base.mjs",
  "scripts": {
    "clean": "rm -rf dist && mkdir dist",
    "rollup": "../../node_modules/rollup/bin/rollup -c ../../scripts/rollup.umd.js && ../../node_modules/rollup/bin/rollup -c ../../scripts/rollup.es.js",
    "build": "yarn run rollup"
  },
  "files": [
    "dist",
    "README.md"
  ],
  "dependencies": {
    "global": "^4.3.2",
    "ncu": "^0.2.1",
    "npm-check-updates": "^2.13.0",
    "polythene-core": "^1.0.0-rc.8"
  },
  "devDependencies": {
    "mithril": "1.1.5",
    "react": "16.0.0",
    "react-dom": "16.0.0",
    "react-hyperscript": "3.0.0"
  },
  "author": "Arthur Clemens <arthurclemens@gmail.com> (http://arthurclemens.com)",
  "license": "MIT"
}
