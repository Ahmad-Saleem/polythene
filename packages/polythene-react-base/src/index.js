export { keys } from "./keys";
export { renderer } from "./renderer";
export { statefulComponent } from "./statefulComponent";
export { viewComponent } from "./viewComponent";
export { Toggle } from "./Toggle";
export { MatchingMithril } from "./MatchingMithril";
