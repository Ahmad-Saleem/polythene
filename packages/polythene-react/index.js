export * from "polythene-react-base";
export * from "polythene-react-button";
export * from "polythene-react-card";
export * from "polythene-react-checkbox";
export * from "polythene-react-dialog";
export * from "polythene-react-dialog-pane";
export * from "polythene-react-fab";
export * from "polythene-react-icon";
export * from "polythene-react-icon-button";
export * from "polythene-react-list";
export * from "polythene-react-list-tile";
export * from "polythene-react-menu";
export * from "polythene-react-radio-button";
export * from "polythene-react-raised-button";
export * from "polythene-react-ripple";
export * from "polythene-react-shadow";
export * from "polythene-react-svg";
export * from "polythene-react-switch";
export * from "polythene-react-textfield";
export * from "polythene-react-toolbar";
