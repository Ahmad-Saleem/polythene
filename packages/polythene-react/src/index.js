export * from "./core";
export { button } from "./button";
export { icon } from "./icon";
export { ripple } from "./ripple";
export { shadow } from "./shadow";
export { svg } from "./svg";
