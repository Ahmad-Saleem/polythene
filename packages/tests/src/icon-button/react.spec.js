import { runSnapshots } from "../../scripts/react-snapshots";
import { renderer, iconButton } from "polythene-react";
import { keys } from "polythene-react-base";
import specTests from "./spec-tests.js";
import reactTests from "./tests-react.js";

runSnapshots({
  tests: specTests({ iconButton, renderer, keys }).concat(reactTests),
  renderer
});
