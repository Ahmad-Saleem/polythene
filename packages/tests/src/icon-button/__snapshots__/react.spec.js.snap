exports[`test Child node (icon component) -- dark theme class 1`] = `
<Button
  animateOnTap={false}
  content={
    <div
      className="pe-icon-button__content">
      <Icon
        svg={
          Object {
            "content": <div
              dangerouslySetInnerHTML={
                  Object {
                      "__html": "<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z\"/></svg>",
                    }
              } />,
          }
        } />
    </div>
  }
  parentClassName="pe-button pe-icon-button "
  wash={false} />
`;

exports[`test Child node (icon component) 1`] = `
<Button
  animateOnTap={false}
  content={
    <div
      className="pe-icon-button__content">
      <Icon
        svg={
          Object {
            "content": <div
              dangerouslySetInnerHTML={
                  Object {
                      "__html": "<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z\"/></svg>",
                    }
              } />,
          }
        } />
    </div>
  }
  parentClassName="pe-button pe-icon-button "
  wash={false} />
`;

exports[`test No options 1`] = `
<Button
  animateOnTap={false}
  content={
    <div
      className="pe-icon-button__content" />
  }
  parentClassName="pe-button pe-icon-button "
  wash={false} />
`;

exports[`test Option: class 1`] = `
<Button
  animateOnTap={false}
  class="class-x"
  content={
    <div
      className="pe-icon-button__content" />
  }
  parentClassName="pe-button pe-icon-button "
  wash={false} />
`;

exports[`test Option: className 1`] = `
<Button
  animateOnTap={false}
  className="className-x"
  content={
    <div
      className="pe-icon-button__content" />
  }
  parentClassName="pe-button pe-icon-button "
  wash={false} />
`;

exports[`test Option: compact 1`] = `
<Button
  animateOnTap={false}
  compact={true}
  content={
    <div
      className="pe-icon-button__content">
      <Icon
        svg={
          Object {
            "content": <div
              dangerouslySetInnerHTML={
                  Object {
                      "__html": "<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z\"/></svg>",
                    }
              } />,
          }
        } />
    </div>
  }
  icon={
    Object {
      "svg": Object {
        "content": <div
          dangerouslySetInnerHTML={
                Object {
                      "__html": "<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z\"/></svg>",
                    }
          } />,
      },
    }
  }
  parentClassName="pe-button pe-icon-button pe-icon-button--compact"
  wash={false} />
`;

exports[`test Option: element 1`] = `
<a
  animateOnTap={false}
  content={
    <div
      className="pe-icon-button__content" />
  }
  element="a"
  parentClassName="pe-button pe-icon-button "
  wash={false} />
`;

exports[`test Option: icon 1`] = `
<Button
  animateOnTap={false}
  content={
    <div
      className="pe-icon-button__content">
      <Icon
        svg={
          Object {
            "content": <div
              dangerouslySetInnerHTML={
                  Object {
                      "__html": "<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z\"/></svg>",
                    }
              } />,
          }
        } />
    </div>
  }
  icon={
    Object {
      "svg": Object {
        "content": <div
          dangerouslySetInnerHTML={
                Object {
                      "__html": "<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z\"/></svg>",
                    }
          } />,
      },
    }
  }
  parentClassName="pe-button pe-icon-button "
  wash={false} />
`;

exports[`test Option: id 1`] = `
<Button
  animateOnTap={false}
  content={
    <div
      className="pe-icon-button__content" />
  }
  id="id-x"
  parentClassName="pe-button pe-icon-button "
  wash={false} />
`;

exports[`test Option: ripple (center) 1`] = `
<Button
  animateOnTap={false}
  content={
    <div
      className="pe-icon-button__content">
      <Icon
        svg={
          Object {
            "content": <div
              dangerouslySetInnerHTML={
                  Object {
                      "__html": "<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z\"/></svg>",
                    }
              } />,
          }
        } />
    </div>
  }
  icon={
    Object {
      "svg": Object {
        "content": <div
          dangerouslySetInnerHTML={
                Object {
                      "__html": "<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z\"/></svg>",
                    }
          } />,
      },
    }
  }
  parentClassName="pe-button pe-icon-button "
  ripple={
    Object {
      "center": true,
    }
  }
  wash={false} />
`;

exports[`test Option: style (colors) 1`] = `
<Button
  animateOnTap={false}
  content={
    <div
      className="pe-icon-button__content">
      <Icon
        svg={
          Object {
            "content": <div
              dangerouslySetInnerHTML={
                  Object {
                      "__html": "<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z\"/></svg>",
                    }
              } />,
          }
        } />
    </div>
  }
  icon={
    Object {
      "svg": Object {
        "content": <div
          dangerouslySetInnerHTML={
                Object {
                      "__html": "<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z\"/></svg>",
                    }
          } />,
      },
    }
  }
  parentClassName="pe-button pe-icon-button "
  style={
    Object {
      "backgroundColor": "#4E342E",
      "color": "#FFCCBC",
    }
  }
  wash={false} />
`;

exports[`test Option: tabindex 1`] = `
<Button
  animateOnTap={false}
  content={
    <div
      className="pe-icon-button__content" />
  }
  parentClassName="pe-button pe-icon-button "
  tabIndex={3}
  wash={false} />
`;

exports[`test Option: type (large) 1`] = `
<Button
  animateOnTap={false}
  content={
    <div
      className="pe-icon-button__content">
      <Icon
        svg={
          Object {
            "content": <div
              dangerouslySetInnerHTML={
                  Object {
                      "__html": "<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z\"/></svg>",
                    }
              } />,
          }
        }
        type="large" />
    </div>
  }
  icon={
    Object {
      "svg": Object {
        "content": <div
          dangerouslySetInnerHTML={
                Object {
                      "__html": "<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z\"/></svg>",
                    }
          } />,
      },
      "type": "large",
    }
  }
  parentClassName="pe-button pe-icon-button "
  wash={false} />
`;

exports[`test Option: type (medium) 1`] = `
<Button
  animateOnTap={false}
  content={
    <div
      className="pe-icon-button__content">
      <Icon
        svg={
          Object {
            "content": <div
              dangerouslySetInnerHTML={
                  Object {
                      "__html": "<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z\"/></svg>",
                    }
              } />,
          }
        }
        type="medium" />
    </div>
  }
  icon={
    Object {
      "svg": Object {
        "content": <div
          dangerouslySetInnerHTML={
                Object {
                      "__html": "<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z\"/></svg>",
                    }
          } />,
      },
      "type": "medium",
    }
  }
  parentClassName="pe-button pe-icon-button "
  wash={false} />
`;

exports[`test Option: type (regular) 1`] = `
<Button
  animateOnTap={false}
  content={
    <div
      className="pe-icon-button__content">
      <Icon
        svg={
          Object {
            "content": <div
              dangerouslySetInnerHTML={
                  Object {
                      "__html": "<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z\"/></svg>",
                    }
              } />,
          }
        }
        type="regular" />
    </div>
  }
  icon={
    Object {
      "svg": Object {
        "content": <div
          dangerouslySetInnerHTML={
                Object {
                      "__html": "<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z\"/></svg>",
                    }
          } />,
      },
      "type": "regular",
    }
  }
  parentClassName="pe-button pe-icon-button "
  wash={false} />
`;

exports[`test Option: type (small) 1`] = `
<Button
  animateOnTap={false}
  content={
    <div
      className="pe-icon-button__content">
      <Icon
        svg={
          Object {
            "content": <div
              dangerouslySetInnerHTML={
                  Object {
                      "__html": "<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z\"/></svg>",
                    }
              } />,
          }
        }
        type="small" />
    </div>
  }
  icon={
    Object {
      "svg": Object {
        "content": <div
          dangerouslySetInnerHTML={
                Object {
                      "__html": "<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z\"/></svg>",
                    }
          } />,
      },
      "type": "small",
    }
  }
  parentClassName="pe-button pe-icon-button "
  wash={false} />
`;

exports[`test Option: wash (true) -- dark theme class 1`] = `
<Button
  animateOnTap={false}
  content={
    <div
      className="pe-icon-button__content">
      <Icon
        svg={
          Object {
            "content": <div
              dangerouslySetInnerHTML={
                  Object {
                      "__html": "<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z\"/></svg>",
                    }
              } />,
          }
        } />
    </div>
  }
  icon={
    Object {
      "svg": Object {
        "content": <div
          dangerouslySetInnerHTML={
                Object {
                      "__html": "<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z\"/></svg>",
                    }
          } />,
      },
    }
  }
  parentClassName="pe-button pe-icon-button "
  wash={true} />
`;

exports[`test Option: wash (true) 1`] = `
<Button
  animateOnTap={false}
  content={
    <div
      className="pe-icon-button__content">
      <Icon
        svg={
          Object {
            "content": <div
              dangerouslySetInnerHTML={
                  Object {
                      "__html": "<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z\"/></svg>",
                    }
              } />,
          }
        } />
    </div>
  }
  icon={
    Object {
      "svg": Object {
        "content": <div
          dangerouslySetInnerHTML={
                Object {
                      "__html": "<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z\"/></svg>",
                    }
          } />,
      },
    }
  }
  parentClassName="pe-button pe-icon-button "
  wash={true} />
`;

exports[`test Themed (color and size) -- dark theme class 1`] = `
<Button
  animateOnTap={false}
  className="tests-icon-button-themed-icon-button"
  content={
    <div
      className="pe-icon-button__content">
      <Icon
        svg={
          Object {
            "content": <div
              dangerouslySetInnerHTML={
                  Object {
                      "__html": "<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z\"/></svg>",
                    }
              } />,
          }
        } />
    </div>
  }
  icon={
    Object {
      "svg": Object {
        "content": <div
          dangerouslySetInnerHTML={
                Object {
                      "__html": "<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z\"/></svg>",
                    }
          } />,
      },
    }
  }
  parentClassName="pe-button pe-icon-button "
  wash={false} />
`;

exports[`test Themed (colors and padding) 1`] = `
<Button
  animateOnTap={false}
  className="tests-icon-button-themed-icon-button"
  content={
    <div
      className="pe-icon-button__content">
      <Icon
        svg={
          Object {
            "content": <div
              dangerouslySetInnerHTML={
                  Object {
                      "__html": "<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z\"/></svg>",
                    }
              } />,
          }
        } />
    </div>
  }
  icon={
    Object {
      "svg": Object {
        "content": <div
          dangerouslySetInnerHTML={
                Object {
                      "__html": "<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z\"/></svg>",
                    }
          } />,
      },
    }
  }
  parentClassName="pe-button pe-icon-button "
  wash={false} />
`;
